### PR TITLE
Cloudwatch

### PR DESCRIPTION
**Pull request by freeformz**
Add support for sending logs to AWS Cloudwatch logs.

This is done by specifying a logs-url in the following format:
```
https://logs.<AMAZON_REGION>.amazonaws.com/<log group name>/<log stream name>
```

Logs are sent as raw unicode text w/o conversion to RFC5424 frames.
Even though log-shuttle can have more than one http outlet at a time, the way the cloudwatch logs service works limits log-shuttle to only sending a single batch at a time. This works even when the number of outlets is > 1.

This is an initial implementation, needs more testing, error handling, etc added to it.

This adds an optional interface that a formatter can support to process the response to a successful request. This is required by cloudwatch logs because each new requests needs a token from the response to the previous request.

This also sets the groundwork for moving the kinesis formatter over to using parts of the aws-go package instead of bmizerany/aws4, which would allow sharing AWS configuration with the aws-go sdk.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 tags
 ./log-shuttle
+log-shuttle.exe
```
Comment (amerine) [inline .gitignore]: cute

File: cloudwatchlogs_formatter.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+)
+
+const (
+	xAmazonTarget             = "Logs_20140328.PutLogEvents"
+	xAmazonJSON11             = "application/x-amz-json-1.1"
+	cloudWatchLogsServiceName = "logs"
```
Comment (amerine) [inline cloudwatchlogs_formatter.go]: Took a look at the docs (https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html) LGTM

File: cmd/log-shuttle/internal/determine_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	for _, tc := range []struct {
+		input    string
+		expected string
+		err      bool
+	}{
+		{input: "foo.bar.com", expected: "", err: true},
+		{input: "", expected: "", err: true},
+		{input: "logs.foo.amazonaws.com", expected: "foo", err: false},
+		{input: "logs.us-east-2.amazonaws.com", expected: "us-east-2", err: false},
+		{input: "logs.us-east-2.amazonaws.com.cn", expected: "us-east-2", err: false},
```
Comment (amerine) [inline cmd/log-shuttle/internal/determine_test.go]: What happens if someone targets an EC2 node like: `ec2-54-203-136-119.us-west-2.compute.amazonaws.com`? Would the "failure to determine the service so we default to the logplex formatter" behavior work?

Comment (freeformz) [reply cmd/log-shuttle/internal/determine_test.go]: It would. The host name you provided isn't a valid ec2 service hostname.

Comment (freeformz) [reply cmd/log-shuttle/internal/determine_test.go]: I added a test to prove it.

**Pull request merged**